homeassistant/components/tuya/event.py

```diff
@@ -17,67 +22,97 @@
 from .const import TUYA_DISCOVERY_NEW, DeviceCategory, DPCode, DPType
 from .entity import TuyaEntity
 from .models import find_dpcode
+from .util import get_dpcode
+
+
+@dataclass(frozen=True)
+class TuyaEventEntityDescription(EventEntityDescription):
+    """Describes Tuya event entity."""
+
+    event_type_conversion: Callable[[Any], str] | None = None
+    event_attributes_conversion: Callable[[Any], dict[str, Any]] | None = None
```

Comment:
I am currently in the process of implementing standard `DPCodeWrapper`

See #155847, #155905 and #156039

I think this should instead be migrated to some kind of new `DPCodeB64DecodeWrapper`

Can we keep this in draft until at least #156039 is merged, and current `event` platform is migrated to use the new `DPCodeEnumWrapper`?

Reply:
Actually we can ignore #156039 - we can rely on #156127 being merged first.

Reply:
The idea is to create the `dpcode_wrapper` BEFORE creating the entity.

We can move this to a local `_get_dpcode_wrapper` function (there will need to be something similar in binary sensor to decide which dpcode_wrapper to use but it's not ready yet: simple True/False or Bitmap key reader or check value in set).

```python
                    if (
                        dpcode_wrapper := _get_dpcode_wrapper(
                            device, description
                        )
                    )
```